sources/rottweiler/src/mount_point.rs

```diff
@@ -0,0 +1,64 @@
+use snafu::prelude::*;
+use std::os::unix::fs::MetadataExt;
+use std::path::{Path, PathBuf};
+
+type Result<T> = std::result::Result<T, snafu::Whatever>;
+
+/// Convert device number from stat encoding to kernel encoding
+///
+/// stat() returns dev_t as u64 using old encoding: (major & 0xff) << 8 | (minor & 0xff) | ((minor & 0xfff00) << 12)
+/// Kernel uses new encoding internally: (major << 20) | minor, where major fits in 12 bits and minor in 20 bits
+/// This function extracts major/minor from old encoding and re-encodes to match kernel's format.
+fn dev_to_kernel_encoding(stat_dev: u64) -> u32 {
+    // Extract major and minor from old stat encoding
+    let major = ((stat_dev >> 8) & 0xff) as u32; // 8 bits max
+    let minor = ((stat_dev & 0xff) | ((stat_dev >> 12) & 0xfff00)) as u32; // 20 bits max
+
+    // Re-encode using kernel format: major in bits 20-31, minor in bits 0-19
+    (major << 20) | minor
+}
+
+/// Represents a filesystem mount point with its path
+pub struct MountPoint {
+    pub path: PathBuf,
+}
+
+impl MountPoint {
+    /// Find the mount point for a given path by walking up the directory tree
+    pub fn from_path(path: &Path) -> Result<Self> {
+        let mut current = path.canonicalize().with_whatever_context(|_| {
+            format!("failed to canonicalize path '{}'", path.display())
+        })?;
+
+        let current_stat = std::fs::metadata(&current).with_whatever_context(|_| {
+            format!("failed to get metadata for '{}'", current.display())
+        })?;
+        let current_dev = dev_to_kernel_encoding(current_stat.dev());
```

Comment:
Do we actually want to do this encoding transformation? We're always comparing devices as returned by rust's stdlib, so the kernel encoding should not matter.

I'm pretty sure we can remove the `dev_to_kernel_encoding` function.